malloc.c

```diff
@@ -179,6 +179,120 @@ void *RETRACE_IMPLEMENTATION(realloc)(void *ptr, size_t size)
 
 RETRACE_REPLACE(realloc, void *, (void *ptr, size_t size), (ptr, size))
 
+void *RETRACE_IMPLEMENTATION(memcpy)(void *dest, const void *src, size_t n)
+{
+	struct rtr_event_info event_info;
+	unsigned int parameter_types[] = {PARAMETER_TYPE_POINTER, PARAMETER_TYPE_POINTER, PARAMETER_TYPE_INT, PARAMETER_TYPE_END};
+	void const *parameter_values[] = {&dest, &src, &n};
+
+	void *p = NULL;
+
+	int use_memmove = 0;
+
+	memset(&event_info, 0, sizeof(event_info));
+	event_info.function_name = "memcpy";
+	event_info.parameter_types = parameter_types;
+	event_info.parameter_values = (void **) parameter_values;
+	event_info.return_value_type = PARAMETER_TYPE_POINTER;
+	event_info.return_value = &p;
+
+	retrace_log_and_redirect_before(&event_info);
+
+	/* check overlapped memory copying */
+	if (abs(dest - src) < n) {
+		use_memmove = 1;
+		event_info.extra_info = "redirected : memmove() (overlapped)";
+	}
+
+	if (!use_memmove)
```

Comment:
I think it is a bad idea. Show that 'memcpy' has overlapping is good point, but change the syscall... It change behaviour of program.

My point here that trace utilite should show the bug but don't try to solve it.